public/pages/main/Main.tsx

```diff
@@ -23,42 +23,58 @@ import { DefineDetector } from '../DefineDetector/containers/DefineDetector';
 import { ConfigureModel } from '../ConfigureModel/containers/ConfigureModel';
 import { DashboardOverview } from '../Dashboard/Container/DashboardOverview';
 import { CoreServicesConsumer } from '../../components/CoreServices/CoreServices';
-import { CoreStart } from '../../../../../src/core/public';
+import { CoreStart, MountPoint } from '../../../../../src/core/public';
 import { AnomalyDetectionOverview } from '../Overview';
+import { getURLQueryParams } from '../DetectorsList/utils/helpers';
 
 enum Navigation {
   AnomalyDetection = 'Anomaly detection',
   Dashboard = 'Dashboard',
   Detectors = 'Detectors',
 }
 
-interface MainProps extends RouteComponentProps {}
+interface MainProps extends RouteComponentProps {
+  setHeaderActionMenu: (menuMount: MountPoint | undefined) => void;
+}
 
 export function Main(props: MainProps) {
+  const { setHeaderActionMenu } = props;
+
   const hideSideNavBar = useSelector(
     (state: AppState) => state.adApp.hideSideNavBar
   );
 
   const adState = useSelector((state: AppState) => state.ad);
   const totalDetectors = adState.totalDetectors;
-  const errorGettingDetectors = adState.errorMessage;
-  const isLoadingDetectors = adState.requesting;
+  const queryParams = getURLQueryParams(props.location);
+  const dataSourceId = queryParams.dataSourceId ? queryParams.dataSourceId : '';
+  const existingParams = "from=0&size=20&search=&indices=&sortField=name&sortDirection=asc";  
+  const constructHrefWithDataSourceId = (basePath, existingParams, dataSourceId) => {
+    const fullUrl = `${window.location.origin}${basePath}?${existingParams}`;
+    const url = new URL(fullUrl);
+    url.searchParams.set('dataSourceId', dataSourceId); 
+    
+    return `#${url.pathname}${url.search}`;
+  };
+  const dashboardHref = dataSourceId ? `#${APP_PATH.DASHBOARD}?dataSourceId=${dataSourceId}` : `#${APP_PATH.DASHBOARD}`;
+  const overviewHref = dataSourceId ? `#${APP_PATH.OVERVIEW}?dataSourceId=${dataSourceId}` : `#${APP_PATH.OVERVIEW}`;
```

Comment:
can we remove these entirely and just use the fn you already define on line 52?

Reply:
Suppose tomorrow the dashboard and overview pages get optional params added, now this will remove those. The fn you define is more flexible.

Reply:
sure, let me update this in the next revision

Reply:
Looks like this wasn't updated, could you remove?